docs/assignment3.md

```diff
@@ -1 +1,418 @@
+* _snippetManager: SnippetManager_
+    * the snippet manager used by the current instance of the app.
+* _cliUI: CliUI_
+    * the cliUI used by the current instance of the app.
+#### Operations
+* _main(args:String[*]): void_
+    * Instantiates SnippetManager, CliUI and JsonIO. After this, CliUI handles the flow of the app.
+#### Associations
+* _Main > JsonIO_
+    * instantiates.
+* _Main > SnippetManager_
+    * instantiates.
+* _Main > CliUI_
+    * instantiates.
+
+### class: GistsApi
+
+This class is responsible for the integration of Githubs Gists with our application
+#### Attributes
+* _httpClient: String: HttpClient_
+    * HttpClient object.
+* _authorization: string_
+    * authorization token that github uses.
+* _usesGithub: boolean_
+    * boolean that shows wether the user has linked his github or not.
+#### Operations
+* _SnippetManager(pathToSnippoDir:String): void_
+    * Constructor function
+* _getInstance(): GistsApi_
+    * returns instance of the GistsApi singleton.
+* _setAuthorization(String providedAuthorization): Void_
+    * set the authorization token to the one the users has provided on login.
+* _setUsesGithub(boolean providedUsesGithub): Void_
+    * boolean that shows wether or not the user is using github gists integration
+* _getUsesGithubGists(): Boolean_
+    * getter function for the usesGithub variable
+* _getAllSnippets(): List<Snippet>_
+    * returns all the gists and parses them to match our snippets
+* _getSpecificSnippet(String gistsId): Snippet_
+    * returns a specific gist with passed gistId
+* _postSpecificSnippet(Snippet snippet): String_
+    * posts a snippet to gist
+* _patchSpecificSnippet(Snippet snippet): Void_
+    * updates a specific snippet to gist
+* _deleteSpecificSnippet(String gistId): Void_
+    * deletes a specific snippet from gist
+* _sendHttpRequest(HttpRequest request):HttpResponse<String>
+    * send a http request, used to send the request to the Gists api.
+* _buildHttpRequest(String method, String url, String body): HttpRequest_
+    * build a http request, used to build the reqeusts we send to the Gists api
+* _onException(Exception e): Void_
+    * handles errors.
+#### Associations
+* GistsApi > SnippetManager 
+    *  provides api.
+* GistsApi > snippet
+    * provides api.
+* GistsApi > JSonIo
+    * provides api.
+## Object diagrams								
+Author(s): Yael Goede
+
+![snapshot Diagram](snapshotDiagram.png)
+In the diagram above a snapshot of the system is shown where the database class is filled with 4 snippets. All snippets have a unique id which makes them unique from each other. Further more all fields are filled with the required information, such as the language, tags and content field. 
+The content field contains the actual code used as snippet by the user. The language field specifies the programming language which is valid for the content field. This field enables features such as the text highlighting feature, which is language specific.
+
```

Comment:
-Good on removing SnippetDB.
-Not that good for not giving different names to different instances of the snippet class.